physical/mysql/mysql.go

```diff
@@ -251,14 +256,18 @@ func NewMySQLClient(conf map[string]string, logger log.Logger) (*sql.DB, error)
 	}
 
 	dsnParams := url.Values{}
-	tlsCaFile, ok := conf["tls_ca_file"]
-	if ok {
+	tlsCaFile, tls_ok := conf["tls_ca_file"]
+	if tls_ok {
 		if err := setupMySQLTLSConfig(tlsCaFile); err != nil {
 			return nil, errwrap.Wrapf("failed register TLS config: {{err}}", err)
 		}
 
 		dsnParams.Add("tls", mysqlTLSKey)
 	}
+	_, pt_ok := conf["plaintext_connection_allowed"]
+	if !(pt_ok || tls_ok) {
```

Comment:
Aha, I'm glad you asked. It's a DeMorgan's Law transformation, it's doing exactly what you wrote. Essentially, it checks to see if either of those are true and inverts the result... which is the same thing as checking if both of those are false.

You can read more here: https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Reply:
Ah cool! Thanks! Looking again made me realize that these are snake case. If it's easy to change them to camel case here, it would be nice, but non-blocker.